src/js/components/Drop/StyledDrop.js

```diff
@@ -27,6 +27,96 @@ const dropKeyFrames = keyframes`
   }
 `;
 
+const isObject = object => object != null && typeof object === 'object';
+
+const deepEqual = (object1, object2) => {
+  const keys1 = Object.keys(object1);
+  const keys2 = Object.keys(object2);
+
+  if (keys1.length !== keys2.length) {
+    return false;
+  }
+
+  for (let i = 0; i < keys1.length; i += 1) {
+    const val1 = object1[keys1[i]];
+    const val2 = object2[keys1[i]];
+    const areObjects = isObject(val1) && isObject(val2);
+    if (
+      (areObjects && !deepEqual(val1, val2)) ||
+      (!areObjects && val1 !== val2)
+    ) {
+      return false;
+    }
+  }
+  return true;
+};
+
+const marginStyle = (theme, align, data, responsive, margin) => {
```

Comment:
Recommending for clarity we adjust the variable/argument names of this function:

margin --> marginProp (since this is the prop a caller has passed)
marginProp --> margin (since this is just the value from theme)
MarginSide --> adjustedMargin (camelCase and implies that we've adjusted the value)